plugins/broadcast/server/server.go

```diff
@@ -0,0 +1,165 @@
+package server
+
+import (
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/tangle"
+	"go.uber.org/atomic"
+
+	"github.com/iotaledger/hive.go/events"
+	"github.com/iotaledger/hive.go/logger"
+	"github.com/iotaledger/hive.go/netutil/buffconn"
+
+	"fmt"
+	"io"
+	"net"
+	"strings"
+)
+
+type Connection struct {
+	bufferedConn *buffconn.BufferedConnection
+	log          *logger.Logger
+	active       bool
+}
+
+var (
+	connectionList      = [256]Connection{}
+	connectionListMutex sync.RWMutex
+	index               atomic.Uint32
+)
+
+// Listen starts a TCP listener and starts a Connection for each accepted connection
+func Listen(bindAddress string, log *logger.Logger, shutdownSignal <-chan struct{}) error {
+
+	listener, err := net.Listen("tcp", bindAddress)
+	if err != nil {
+		return fmt.Errorf("failed to start Broadcast daemon: %w", err)
+	}
+
+	go func() {
+		for {
+			connection, err := listener.Accept()
+			if err != nil {
+				return
+			}
+			log.Infof("Started connection: %s", connection.RemoteAddr().String())
+			go handleConnection(connection, log, shutdownSignal)
+		}
+	}()
+
+	go func() {
+		defer func(listener net.Listener) {
+			err := listener.Close()
+			if err != nil {
+				log.Infof("Error closing listener: %s", err)
+			}
+		}(listener)
+
+		<-shutdownSignal
+
+		log.Infof("Closing Broadcast server...")
+		idx := int(index.Load())
+		connectionListMutex.Lock()
+		for i := 0; i < idx; i++ {
+			connectionList[i].active = false
+		}
+		connectionListMutex.Unlock()
+		log.Infof("Closing Broadcast server... \tDone")
+	}()
+
+	return nil
+}
+
+func handleConnection(conn net.Conn, log *logger.Logger, shutdownSignal <-chan struct{}) {
+	connectionListMutex.Lock()
+	defer connectionListMutex.Unlock()
+
+	idx := int(index.Load())
+	connectionList[idx] = Connection{
+		bufferedConn: buffconn.NewBufferedConnection(conn, tangle.MaxMessageSize),
+		log:          log,
+		active:       true,
+	}
+	index.Inc()
+
+	bufferedConnDataReceived, bufferedConnClosed := connectionList[idx-1].readLoop()
+
+	select {
+	case data := <-bufferedConnDataReceived:
+		//No input required. For debugging it will be printed
+		connectionList[idx-1].log.Infof("Data received:%v", data)
+		connectionList[idx-1].active = false
+	case <-shutdownSignal:
+		connectionList[idx-1].log.Infof("Shutdown signal received")
```

Comment:
Why does the shutdown signal trigger this for loop? Won't you end up doing the same for X amount of connections while the "shutdown"? Perhaps this is simply a miss copy-and-paste from the cleanup code from `Listen()`?